app/presenters/hyrax/displays_image.rb

```diff
@@ -45,13 +44,18 @@ def image_format(channels)
         channels&.include?('rgba') ? 'png' : 'jpg'
       end
 
-      def lookup_original_file_id
-        result = original_file_id
-        if result.blank?
-          Rails.logger.warn "original_file_id for #{id} not found, falling back to Fedora."
-          result = Hyrax::VersioningService.versioned_file_id ::FileSet.find(id).original_file
-        end
-        result
+      def latest_file_id
+        @latest_file_id ||=
+          begin
+            result = original_file_id
+
+            if result.blank?
+              Rails.logger.warn "original_file_id for #{id} not found, falling back to Fedora."
+              result = Hyrax::VersioningService.versioned_file_id ::FileSet.find(id).original_file
```

Comment:
I know this is a port, but line 54 concerns me. On a quick read with line breaks I read it as:

`Hyrax::VersioningService.versioned_file_id::FileSet.find(id).original_file`

However, in looking at a wider buffer, I see it's actually semantically this: `Hyrax::VersioningService.versioned_file_id(::FileSet.find(id).original_file)`

I again would prefer a local variable `original_file = ::FileSet.find(id)` then pass that to the poorly named `#versioned_file_id` method. I'd go with `find_versioned_file_id_for` as the method name.

However, that is certainly not a refactor that need be done in this PR. Just an observation while reviewing.

Reply:
agreed. this `displays_image` refactor is something i tried to do in as fixed a way as possible.

i'm not deeply confident in the code in this module (and it's not unit tested) so i've tried to limit the scope of changes.